packages/ember-data/lib/system/store.js

```diff
@@ -1522,15 +1522,53 @@ Store = Service.extend({
   /**
     Push some data for a given type into the store.
 
-    This method expects normalized data:
+    This method expects [JSON API](http://jsonapi.org/) compound document with [your primary data wrapped inside `data` property](http://jsonapi.org/format/#document-top-level):
```

Comment:
Can we add that this method expects a "normalized JSON API Document" (as in, camelCased members and singular dasherized types).